src/infill.cpp

```diff
@@ -551,40 +507,26 @@ void Infill::generateZigZagInfill(Polygons& result, const coord_t line_distance,
  * Edit: the term scansegment is wrong, since I call a boundary segment leaving from an even scanline to the left as belonging to an even scansegment, 
  *  while I also call a boundary segment leaving from an even scanline toward the right as belonging to an even scansegment.
  */
-void Infill::generateLinearBasedInfill(const int outline_offset, Polygons& result, const int line_distance, const PointMatrix& rotation_matrix, ZigzagConnectorProcessor& zigzag_connector_processor, const bool connected_zigzags, coord_t extra_shift)
+void Infill::generateLinearBasedInfill(Polygons& result, const int line_distance, const PointMatrix& rotation_matrix, ZigzagConnectorProcessor& zigzag_connector_processor, const bool connected_zigzags, coord_t extra_shift)
 {
     if (line_distance == 0)
     {
         return;
     }
-    if (outer_contour.size() == 0)
+    if (inner_contour.empty())
     {
         return;
     }
 
     coord_t shift = extra_shift + this->shift;
 
-    if (outline_offset != 0 && perimeter_gaps)
-    {
-        const Polygons gaps_outline =
-            outer_contour.offset(outline_offset + infill_line_width / 2 + perimeter_gaps_extra_offset);
-        perimeter_gaps->add(outer_contour.difference(gaps_outline));
-    }
-
-    Polygons outline = outer_contour.offset(outline_offset + infill_overlap);
-
+    Polygons outline = inner_contour; //Make a copy. We'll be rotating this outline to make intersections always horizontal, for better performance.
     if (outline.size() == 0)
     {
         return;
     }
-    //TODO: Currently we find the outline every time for each rotation.
-    //We should compute it only once and rotate that accordingly.
-    //We'll also have the guarantee that they have the same size every time.
-    //Currently we assume that the above operations are all rotation-invariant,
-    //which they aren't if vertices fall on the same coordinate due to rounding.
-    crossings_on_line.resize(outline.size()); //One for each polygon.
-
     outline.applyMatrix(rotation_matrix);
+    crossings_on_line.resize(outline.size()); //One for each polygon.
 
     if (shift < 0)
```

Comment:
I don't think this really improves the readability here though. The syntax is a bit... clumsy. It's much easier to read just that we're setting `shift` to be some sum and then perform some minor calculations on it just line by line. To change it to this just because `shift` can be `const` for the second half of the function is a bit overkill.

I also like the `if` `else` better since it semantically denotes two distinct cases, rather than this technique you propose where it looks like a bit extra being done in one case and the other half being done in both cases.

I'll move the declaration of the variable down lower though. That should indeed be clustered together.